src/main/java/com/yegor256/tojos/MnCsv.java

```diff
@@ -183,7 +183,8 @@ public void close() {
      */
     private static Collection<Map<String, String>> dup(final Collection<Map<String, String>> rows) {
         final Collection<Map<String, String>> list = new ArrayList<>(rows.size());
-        for (final Map<String, String> map : rows) {
+        final List<Map<String, String>> snapshot = new ArrayList<>(rows);
+        for (final Map<String, String> map : snapshot) {
```

Comment:
@VladMilash Do we need the `snapshot` variable here? 
This might help: https://www.yegor256.com/2015/09/01/redundant-variables-are-evil.html